src/trans/gpu/algor/hicblas_gemm.hip.cpp

```diff
@@ -124,28 +124,21 @@ void run_group_graph(Gemm &&gemm, int resol_id, int m, const int *n,
   auto graph = graphCache.find(key);
   if (graph == graphCache.end()) {
     // this graph does not exist yet
-    hipStream_t stream;
-    HIC_CHECK(hipStreamCreate(&stream));
-
-    hipGraph_t new_graph;
-    hipGraphCreate(&new_graph, 0);
+    hipStream_t captureStream;
+    HIC_CHECK(hipStreamCreate(&captureStream));
+    HIC_CHECK(hipStreamBeginCapture(captureStream, hipStreamCaptureModeGlobal));
     for (int i = 0; i < batchCount; ++i) {
       if (m == 0 || n[i] == 0 || k[i] == 0)
         continue;
 
-      HIC_CHECK(hipStreamBeginCapture(stream, hipStreamCaptureModeGlobal));
-      gemm(stream, m, n[i], k[i], alpha, A + offsetsA[i], lda, B + offsetsB[i],
+      gemm(captureStream, m, n[i], k[i], alpha, A + offsetsA[i], lda, B + offsetsB[i],
            ldb[i], beta, C + offsetsC[i], ldc);
-      hipGraph_t my_graph;
-      HIC_CHECK(hipStreamEndCapture(stream, &my_graph));
-      hipGraphNode_t my_node;
-      HIC_CHECK(
-          hipGraphAddChildGraphNode(&my_node, new_graph, nullptr, 0, my_graph));
```

Comment:
Again it looks like these changes the graph capturing behaviour quite a lot. Is this strictly necessary for amdflang?